### PR TITLE
Fix #2679: remove-exploration-java-proto-lite from the model library

### LINKED ISSUE #2679: Remove :exploration_java_proto_lite from the model library
This task is part of the build system migration from Gradle to Bazel. A prerequesite before working on this issue is that you are able to build Oppia using Bazel. Check out [Oppia Bazel guide](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Guide) for Bazel terminology and more on what Bazel does.

## 1. Remove the export

In 

`/oppia-android/model/BUILD.bazel`

Remove :exploration_java_proto_lite from the list of exports in the model library

```BUILD
android_library(
    name = "model",
    visibility = ["//visibility:public"],
    exports = [
        ":event_logger_java_proto_lite",
        ":exploration_java_proto_lite",
        ":interaction_object_java_proto_lite",
        ":onboarding_java_proto_lite",
        ":profile_java_proto_lite",
        ":question_java_proto_lite",
        ":subtitled_html_java_proto_lite",
        ":subtitled_unicode_java_proto_lite",
        ":thumbnail_java_proto_lite",
        ":topic_java_proto_lite",
        ":translation_java_proto_lite",
        ":voiceover_java_proto_lite",
    ],
)
```

## 2. Add public visibility to the java_lite_proto_library

Add 

`    visibility = ["//visibility:public"],`

to the java_lite_proto_library that has the name exploration_java_proto_lite.

## 3. Build the app using Bazel

Libraries that are indirectly depending on `exploration_java_proto_lite` will now fail to build.  Try to build the app with Bazel and add the deps dependency

`//model:exploration_java_proto_lite,`

to all the libraries that are now failing to build. You will have to look around for BUILD.bazel files that have libraries that require this replaced dependency.

These libraries will already have the //model dependency and since we removed the :exploration_java_proto_lite from the model library itself, they will need to add the newly available public library directly as a dependency. (Not all libraries with the //model dependency may require the newly available library)

Once the app builds, you're done!

### PR DESCRIPTION
**Pull request by yashraj-01**
## Explanation

Fixes #2679 . Removed `:exploration_java_proto_lite` from the model library. Added `visibility = ["//visibility:public"],` to the `java_lite_proto_library` that has the name `exploration_java_proto_lite`.

The build was successful after Step 1 and Step 2. So, it was not required to follow Step 3.
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by yashraj-01** [comment]
@fsharpasharp @rt4914 PTAL.

**Comment by fsharpasharp** [comment]
It is unexpected to me that the app will build without step 3. Make sure that you are building the entire application.

We'll have to wait for the checks to be fixed and rerun the checks. Everything else looks good.

File: model/BUILD.bazel [hunk 1]
```diff
@@ -186,14 +186,14 @@ format_import_proto_library(
 java_lite_proto_library(
     name = "exploration_java_proto_lite",
     deps = [":exploration_proto"],
+    visibility = ["//visibility:public"],
```
Comment (vinitamurthi) [inline model/BUILD.bazel:200]: In general we should avoid doing this. Any reason why we want to make this public?

Comment (fsharpasharp) [reply model/BUILD.bazel:200]: @vinitamurthi, are you saying that we should be explicit about the visibility?

**Comment by yashraj-01** [comment]
Screenshot from 2021-02-12 22-13-40

@fsharpasharp I am somehow getting this error now. Previously it worked fine but now this error is coming. I also followed the fix provided above but it also does not work. Please help me to fix this error.

**Comment by fsharpasharp** [comment]
Replace ~ with your home directory.

**Comment by yashraj-01** [comment]
> Replace ~ with your home directory.

In which of the following commands should I replace it. I'm currently at **step 4**.
Screenshot from 2021-02-12 22-44-29

**Comment by fsharpasharp** [comment]
In your ~/.bazelrc file

File: model/BUILD.bazel (hunk 1 above)
Comment (BenHenning) [reply model/BUILD.bazel:200]: Does https://github.com/oppia/oppia-android/pull/2699#discussion_r575296434 help clarify the need @vinitamurthi or do you have a separate concern?

**Comment by BenHenning** [comment]
Similar to https://github.com/oppia/oppia-android/pull/2699#issuecomment-778557193 we need verify that all targets are now building with this change. Updating the branch to trigger a full run of all affected Bazel test targets.

**Comment by yashraj-iitr** [comment]
> In your ~/.bazelrc file

Sorry but I'm not able to understand what exactly you mean. Can you please explain a bit more?

**Comment by yashraj-01** [comment]
> It is unexpected to me that the app will build without step 3. Make sure that you are building the entire application.
> 
> We'll have to wait for the checks to be fixed and rerun the checks. Everything else looks good.

Screenshot from 2021-02-13 21-33-36
@fsharpasharp I again ran the build and it is showing build successful. What do you suggest I should do?

**Comment by yashraj-01** [comment]
@BenHenning @vinitamurthi PTAL.

**Comment by fsharpasharp** [comment]
I see that there is one failing check. Try to identify the issue and make sure that the all the tests pass locally as well. Let me know if you get stuck.

Run

```
~/oppia-bazel/bazel test //app/...
```

and

```
~/oppia-bazel/bazel test //...
```

if the previous one passes.

**Comment by yashraj-01** [comment]
> I see that there is one failing check. Try to identify the issue and make sure that the all the tests pass locally as well. Let me know if you get stuck.
> 
> Run
> 
> ```
> ~/oppia-bazel/bazel test //app/...
> ```
> 
> and
> 
> ```
> ~/oppia-bazel/bazel test //...
> ```
> 
> if the previous one passes.

https://github.com/oppia/oppia-android/pull/2722#issuecomment-778800709

I asked @FareesHussain about this failed check and he said it just needs a re-run. I have attached the link of the comment. 

Also, when I ran the test you mentioned, it gets stuck at check number 11,978.

Please, tell me how to proceed now.

**Comment by fsharpasharp** [comment]
Okay, could you try to sync to head and run the checks again? It could help us rule out if it's a problem with the current commit or with the CI.

**Comment by BenHenning** [comment]
Deferring to @fsharpasharp to figure out why we're seeing what appears to be Java proto lite libraries being generated for each proto (including transitive proto dependencies). Please reassign me once this PR is ready for review.

**Comment by BenHenning** [comment]
Ah, didn't meant to close the PR. Checks are re-running now, sorry about that. Old checks can be seen in the commit above, but some might get cancelled by the automatic canceller.

**Comment by yashraj-01** [comment]
@fsharpasharp @BenHenning all the checks have passed. PTAL. Thanks.

**Comment by fsharpasharp** [comment]
I'm going to spend some time to look into where the dependency is getting pulled from. I'll open some similar issues in the meantime, if you are itching to do some more work. Thanks for the patience, @yashraj-iitr.

**Comment by fsharpasharp** [comment]
```BUILD
  format_import_proto_library(                                                                                                                                                                 
      name = "question",                                                                                                                                                                       
      src = "src/main/proto/question.proto",                                                                                                                                                   
      deps = [                                                                                                                                                                                 
          ":exploration_proto",                                                                                                                                                                
          ":subtitled_html_proto",                                                                                                                                                             
          ":subtitled_unicode_proto",                                                                                                                                                          
      ],                                                                                                                                                                                       
  )                                                                                                                                                                                            
                                                                                                                                                                                               
  java_lite_proto_library(                                                                                                                                                                     
      name = "question_java_proto_lite",                                                                                                                                                       
      deps = [":question_proto"],                                                                                                                                                              
  )   
```

`question_java_proto_lite` is indirectly exporting the exploration proto. I think we can go ahead and merge this PR, @BenHenning . When we remove the `question_java_proto_lite` from the exports list of `model` we can add both dependencies back to where they are used.

To be thorough we could remove 

`      visibility = ["//visibility:public"],`

from the `exploration_java_proto_lite` library in this PR. This could make it clear how the dependency propagates.

**Comment by BenHenning** [comment]
Ack, it seems like we can't solve this right now but moving in this direction is still satisfying Bazel's proto best practices.

**Review by BenHenning — APPROVED** [review]
Per my comment above, we'll just keep in mind that protos imply transitive dependencies as we modularize the codebase. We should take care to try and depend on the most specific proto possible for a given context.

Approved since this is solving the original issue. Thanks @yashraj-iitr!

**Pull request merged**